src/app/campaign/store/actions/campaign-action.service.spec.ts

```diff
@@ -45,48 +52,48 @@ describe('CampaignActionService', () => {
       new MockHalDoc({ ...flightDocFixture, id: flightIds[2] }),
       new MockHalDoc({ ...flightDocFixture, id: flightIds[3] })
     ];
-    jest.spyOn(service, 'loadAvailability');
-    fixture.ngZone.run(() => {
-      const loadAction = new campaignActions.CampaignLoadSuccess({ campaignDoc, flightDocs });
-      store.dispatch(loadAction);
+    const flightDaysDocs = {
+      [flightIds[0]]: flightDaysDocFixture,
+      [flightIds[1]]: flightDaysDocFixture,
+      [flightIds[2]]: flightDaysDocFixture,
+      [flightIds[3]]: flightDaysDocFixture
+    };
+    jest.spyOn(service, 'loadFlightPreview');
 
-      router
-        .navigateByUrl(`/campaign/${campaignFixture.id}/flight/${flightFixture.id}`)
-        .then(() => (dispatchSpy = jest.spyOn(store, 'dispatch')));
+    fixture.ngZone.run(() => {
+      router.navigateByUrl(`/campaign/${campaignFixture.id}/flight/${flightFixture.id}`).then(() => {
+        const loadAction = new campaignActions.CampaignLoadSuccess({
+          campaignDoc,
+          flightDocs,
+          flightDaysDocs
+        });
+        store.dispatch(loadAction);
+        dispatchSpy = jest.spyOn(store, 'dispatch');
+      });
     });
   });
 
-  it('should load availability from flight id change', () => {
-    expect(service.loadAvailability).toHaveBeenCalledWith({ MOCKS: {}, ERRORS: {}, ...flightFixture });
-  });
-
-  it('should load availability and allocation preview when flight form is updated', () => {
-    jest.spyOn(service, 'loadAvailabilityAllocationIfChanged');
+  it('should load flight preview when flight form is updated', () => {
+    jest.spyOn(service, 'loadPreviewIfFlightFormChanged');
     service.updateFlightForm({ ...flightFixture, endAt: moment.utc() }, true, true);
-    expect(service.loadAvailabilityAllocationIfChanged).toHaveBeenCalled();
+    expect(service.loadPreviewIfFlightFormChanged).toHaveBeenCalled();
   });
 
-  it('should dispatch action to load allocation preview', () => {
-    const { id: flightId, name, startAt, totalGoal, dailyMinimum, set_inventory_uri, zones } = flightFixture;
+  it('should dispatch action to load flight preview', () => {
     const endAt = moment.utc();
-    const createdAt = new Date();
-
     const formFlight = { ...flightFixture, endAt };
-    const localFlight = { ...flightFixture, createdAt };
-    service.loadAvailabilityAllocationIfChanged(formFlight, localFlight);
-
-    expect(dispatchSpy).toHaveBeenLastCalledWith(
-      new allocationPreviewActions.AllocationPreviewLoad({
-        flightId,
-        createdAt,
-        set_inventory_uri,
-        name,
-        startAt: startAt.toDate(),
-        endAt: endAt.toDate(),
-        totalGoal,
-        dailyMinimum,
-        zones
-      })
```

Comment:
jest seems to have a matcher bug where it fails with "values have no visual difference," so I resorted to using JSON.stringify on the spy.mock.calls instead of toHaveBeenCalledWith. annoying.